src/Utilities/InternetParser.cs

```diff
@@ -1,32 +1,31 @@
 ﻿using System;
-using System.IO;
 using System.Linq;
 using System.Net;
 using System.Net.Http;
 using System.Threading.Tasks;
+using ePiggyWeb.DataManagement.Goals;
 using HtmlAgilityPack;
-
 namespace ePiggyWeb.Utilities
 {
-    public static class InternetParser
+    public class InternetParser
     {
-        /*Example of calling method:
-         * Task.Run(() => InternetParser.GetHTMLAsync()).Wait();
-        */
-        private static readonly string ResourceDirectoryParsedGoal = Directory.GetParent(Environment.CurrentDirectory)
-                                                                         .Parent?.Parent?.FullName +
-                                                                     @"\resources\textData\parsedGoal.txt";
-        public static async Task ReadPriceFromCamel(string itemName)
-        {
-            var HttpClient = new HttpClient();
+        private readonly Func<int, string, bool> _isTooLong = (x, s) => s.Length > x;
+        private readonly Func<decimal, decimal> _convertToEur = (price) => price * (decimal) 1.11;
+        private readonly HttpClient _httpClient;
 
+        public InternetParser(HttpClient httpClient)
+        {
+            _httpClient = httpClient;
+        }
```

Comment:
Nice idea to implement Func, but with the following code we could implement delegates here as well ( not that it makes any sense, but it would fulfill the requirement somewhat :D)

        private delegate decimal Cnv(decimal num);
        private readonly Cnv _cnv;
        public InternetParser(HttpClient httpClient)
        {
            _httpClient = httpClient;
            _cnv = _convertToEur.Invoke;
        }